src/genmsg/msg_loader.py

```diff
@@ -181,8 +182,7 @@ def convert_constant_value(field_type, val):
             raise InvalidMsgSpec("cannot coerce [%s] to %s (out of bounds)"%(val, field_type))
         return val
     elif field_type == 'bool':
-        # TODO: need to nail down constant spec for bool
-        return True if eval(val) else False
+        return ast.literal_eval(val)
```

Comment:
TL;DR I think that the logic for ensuring the return of a boolean should be kept here to preserve backward compatibility, aka `return True if ast.literal_eval(val) else False`.

Details: The new logic will make this function return numbers (integers or float) instead of python `bool` in the case where users provide numbers as a default.
For example the following msg file:
```
bool FOO=42
bool BAR=0
bool BLA=42.42
bool BLABLA=-42.42
````

currently results in  (generated python message):
```
FOO = True
BAR = False
BLA = True
BLABLA = True
```
but with this PR will result in : 
```
FOO = 42
BAR = 0
BLA = 42.42
BLABLA = -42.42
```